packages/entity/src/ComposedEntityCacheAdapter.ts

```diff
@@ -40,11 +44,35 @@ export default class ComposedEntityCacheAdapter<TFields> extends EntityCacheAdap
           newUnfulfilledFieldValues.push(fieldValue);
         } else {
           retMap.set(fieldValue, cacheResult);
+          const fulfilledFieldValues = nullthrows(fulfilledFieldValuesByCacheIndex[i]);
+          fulfilledFieldValues.push(fieldValue);
         }
       }
       unfulfilledFieldValues = newUnfulfilledFieldValues;
     }
 
+    // Recache values from higher layers that were not found in lower layers
+    for (let i = 0; i < this.cacheAdapters.length; i++) {
```

Comment:
we should be able to run a `Promise.all` here instead of awaiting within a for loop since the re-cache operations are logically independent

Reply:
looking at the other code, we've made a point to `await` on caches closest to the database, then move onto the ones closer to the application. I think this was to avoid bugs where the cache closest to the application was stale, but the ones closer to the application was more up-to-date. 

I think the most correct approach would be to iterate backwards (ie) `for (let i = this.cacheAdapters.length - 1; i >= 0; i--) {` to keep this behaviour consistent

Reply:
Ah right. Good call.